internal/retry/retry.go

```diff
@@ -0,0 +1,53 @@
+package retry
+
+import (
+	"fmt"
+	"time"
+)
+
+type Action func() error
+type checkErr func(err error) bool
+
+type retry struct {
+	action   Action
+	checkErr checkErr
+	attempt  uint
+	max      uint
+}
+
+func New(action Action, checkErr checkErr, count uint) *retry {
+	return &retry{
+		attempt:  1,
+		checkErr: checkErr,
+		max:      count,
+		action:   action,
+	}
+}
+
+func (r *retry) Run() error {
+	var err error
+	for r.attempt <= r.max {
+		err = r.action()
+		if err == nil || !r.checkErr(err) || r.attempt == r.max {
```

Comment:
Последнее уже проверяется выше

```suggestion
		if err == nil || !r.checkErr(err) {
```

Я понимаю, что ты добавил эту проверку, чтобы не вызывать `wait` лишний раз, но попробуй упростить код и обойтись здесь без дублирования логики